topo/node/cptx/cptx.go

```diff
@@ -26,18 +39,134 @@ func New(pb *topopb.Node) (node.Implementation, error) {
 
 type Node struct {
 	pb *topopb.Node
+	cliConn *scraplinetwork.Driver
 }
 
 func (n *Node) Proto() *topopb.Node {
 	return n.pb
 }
 
+// WaitCLIReady attempts to open the transport channel towards a Network OS and perform scrapligo OnOpen actions
+// for a given platform. Retries with exponential backoff.
+func (n *Node) WaitCLIReady(sleep time.Duration, attempts int) error {
+    var err error = nil
+    for i := 0; i < attempts; i++ {
+        if i > 0 {
+			log.Debugf("%s - cli not ready - waiting.", n.pb.Name)
+            time.Sleep(sleep)
+            sleep *= 2
+        }
+		err = n.cliConn.Open()
+        if err == nil {
+		    log.Debugf("%s - cli ready.", n.pb.Name)
+	        return nil
+		}
+	}
+    return err
+}
+
+// PatchCLIConnOpen sets the OpenCmd and ExecCmd of system transport to work with `kubectl exec` terminal.
+func (n *Node) PatchCLIConnOpen(ns string) error {
+	t, ok := n.cliConn.Transport.Impl.(scraplitransport.SystemTransport)
+	if !ok {
+		return ErrIncompatibleCliConn
+	}
+
+	t.SetExecCmd("kubectl")
+	t.SetOpenCmd([]string{"exec", "-it", "-n", ns, n.pb.Name, "--", "cli", "-c"})
+
+	return nil
+}
+
+// SpawnCLIConn spawns a CLI connection towards a Network OS using `kubectl exec` terminal and ensures CLI is ready
+// to accept inputs.
+func (n *Node) SpawnCLIConn(ns string) error {
```

Comment:
please add unittest coverage here as well

Reply:
Sure, @marcushines. I will push another patch soon.

Reply:
@marcushines for unit test to work, it needs some changes in scrapligo. I am currently working to fix things in scrapligo. As soon as that is done, I'll push the coverage related patch.

Reply:
@marcushines The latest commit contains unit test coverage but it requires a few fixes in scrapligo. I have opened a PR on scrapligo : scrapli/scrapligo#50
Opening this for review as this can get reviewed in parallel while scrapligo PR is being reviewed.
Config push should work fine but the "go test" is expected to fail as scrapligo fixes are missing.
I will push another commit as soon as scraplogo fixes are in.

Reply:
@marcushines required scrapligo changes are merged and are available as part of scrapligo v0.1.1. I have updated the mod versioning accordingly. Now this pull request is ready for further review and merge. UT coverage is now added.